actionpack/lib/action_controller/test_case.rb

```diff
@@ -593,7 +607,14 @@ def process(action, http_method = 'GET', *args)
           @request.env['RAW_POST_DATA'] = args.shift
         end
 
-        parameters, session, flash = args
+        if args[0] && args[0].keys.any? { |k| REQUEST_KWARGS.include?(k) }
+          parameters = args[0][:params]
+          session =    args[0][:session]
+          flash =      args[0][:flash]
+        else
+          parameters, session, flash = args
```

Comment:
Good point, thanks.

Here is rdoc syntax for usual arguments:

```
- +action+: The controller action to call.
- +http_method+: Request method used to send the http request. Possible values
  are +GET+, +POST+, +PATCH+, +PUT+, +DELETE+, +HEAD+. Defaults to +GET+.
- +parameters+: The HTTP parameters. This may be +nil+, a hash, or a
  string that is appropriately encoded (+application/x-www-form-urlencoded+
  or +multipart/form-data+).
- +session+: A hash of parameters to store in the session. This may be +nil+.
- +flash+: A hash of parameters to store in the flash. This may be +nil+.
```

Any suggestions how it will look like with kwargs? Couldn't find it in docs.